docs/rfcs/rfc-029-unwind-create-auth.md

```diff
@@ -0,0 +1,1401 @@
+
+-   `Roles`
+-   `Bind`
+-   `Where`
+-   `Allow`
+
+`Where` and `Allow` seems to have no impact during creation, for this reason, this solution is focused on the `Bind` and `Roles` rule.
+
+There is a common pattern in how the `@auth` directive could impact the create mutation depending on how it is used in the type definitions and could be resumed as:
+
+-   ### Entity-type auth
+
+    **Schema**
+
+    ```graphql
+    type User {
+        id: ID
+        name: String
+    }
+
+    type Post @auth(rules: [{ operations: [CREATE], roles: ["admin"] }]) {
+        title: String
+        content: String
+        moderators: [User!]! @relationship(type: "MODERATES_POST", direction: IN)
+    }
+    ```
+
+-   ### Entity-type auth - Field-level definition
+
+    **Schema**
+
+    ```graphql
+    type User {
+        id: ID
+        name: String
+    }
+
+    type Post {
+        title: String @auth(rules: [{ operations: [CREATE], roles: ["admin"] }])
+        content: String
+        moderators: [User!]! @relationship(type: "MODERATES_POST", direction: IN)
+    }
+    ```
+
+-   ### Nested Entity-type auth
+
+    **Schema**
+
+    ```graphql
+    type User @auth(rules: [{ operations: [CREATE], roles: ["admin"] }]) {
+        id: ID
+        name: String
+    }
+
+    type Post {
+        title: String
+        content: String
+        moderators: [User!]! @relationship(type: "MODERATES_POST", direction: IN)
+    }
+    ```
+
+-   ### Nested Entity-type auth - Field-level definition
+
+    **Schema**
+
+    ```graphql
+    type User {
+        id: ID @auth(rules: [{ operations: [CREATE], roles: ["admin"] }])
+        name: String
+    }
+
+    type Post {
+        title: String
+        content: String
+        moderators: [User!]! @relationship(type: "MODERATES_POST", direction: IN)
+    }
+    ```
+
+-   ### Particular case: Entity-type Bind - across relationship
+
+    **Schema**
+
+    ```graphql
+    type User {
+        id: ID
+        name: String
+    }
+    type Post @auth(rules: [{ operations: [CREATE], bind: { creator: { id: "$jwt.sub" } } }]) {
+        title: String
+        content: String
+        creator: [User!]! @relationship(type: "HAS_POST", direction: IN)
+    }
+    ```
+
+### Entity-type solution
+
+Entity type `@auth` as well as Nested Entity type `@auth` no needs particular consideration when used in the unwind-create context.
+
+It could be appended directly on the entity modified,
+For instance, preventing a post is created by users without the `ADMIN` role, could be achieved as follow:
+
+```cypher
+UNWIND $create_param0 AS create_var1
+CALL {
+    WITH create_var1
+    CREATE (create_this0:`Post`)
+    SET
+        create_this0.content = create_var1.content
+    WITH create_this0, create_var1
+    CALL apoc.util.validate(NOT (any(auth_var1 IN ["admin"] WHERE any(auth_var0 IN $auth.roles WHERE auth_var0 = auth_var1))), "@neo4j/graphql/FORBIDDEN", [0])
```

Comment:
I would consider using `validatePredicate` instead here - I think it's nicer than having procedure calls spattered about! Consider this comment to be applied across this PR.

```suggestion
    WHERE apoc.util.validatePredicate(NOT (any(auth_var1 IN ["admin"] WHERE any(auth_var0 IN $auth.roles WHERE auth_var0 = auth_var1))), "@neo4j/graphql/FORBIDDEN", [0])
```